src/emulator/controller.ts

```diff
@@ -747,7 +747,10 @@ export async function startAll(
     hubLogger.logLabeled(
       "WARN",
       "emulators",
-      "The Emulator UI requires a project ID to start. Configure your default project with 'firebase use' or pass the --project flag."
+      "The Emulator UI is not starting, either because none of the emulated " +
+        "features interact with the Emulator UI or it cannot determine the " +
```

Comment:
What does "features" mean in this context?

Reply:
is "products" any better? i.e. "hosting", "functions", "auth", "database" - what we can actually emulate

Reply:
yep, "products" is more appropriate.

I'm not sure what "none of the emulated products interact with the Emulator UI" means, though?  Aren't all the products that work with the Emulator Suite also available in the Emulator UI?

Reply:
wordsmith'd